### ISSUE TITLE
_getrenvvars does not handle envvars with newlines

### ISSUE THREAD

**Issue by collares** [rpy2/rpy2#1066]
**Describe the issue or bug**

`_getrenvvars`, introduced in #1037, [invokes R to generate a CSV with environment variables](https://github.com/rpy2/rpy2/blob/f8770064227b2ea5695d8627a4d1476f6d39361a/rpy2/rinterface.py#L1131), splits lines according to `os.linesep`, and passes the result along to `csv.reader`. This assumes that the contents of `os.linesep` (i.e. newlines) do not appear in any environment variable. In an automation run, we had two environment variables which did contain linebreaks, and the CSV exported by R as part of `_getrenvvars` contained the lines

```
"postFixup","wrapPythonPrograms
"
"postPatch","substituteInPlace 'rpy2/rinterface_lib/embedded.py' --replace '@NIX_R_LIBS_SITE@' \"$R_LIBS_SITE\"
substituteInPlace 'requirements.txt' --replace 'pytest' \"\"
"
```

I think the variable contents are not important, other than containing multiple newlines. They cause the later line `for k, v in reader:` in `_getrenvvars` to fail (see error below).

**To Reproduce**

Create an environment variable with values containing multiple newlines, and run rpy2 tests.

**Expected behavior**

rpy2 tests pass

**Error**

```
________ ERROR collecting rpy2/tests/rinterface/test_bufferprotocol.py _________
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/python.py:531: in collect
    self._inject_setup_module_fixture()
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/python.py:310: in obj
    self._obj = obj = self._getobj()
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/python.py:528: in _getobj
    return self._importtestmodule()
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/pathlib.py:567: in import_path
    importlib.import_module(module_name)
/nix/store/ffll6glz3gwx342z0ch8wx30p5cnqz1z-python3-3.11.5/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
/nix/store/lhjsx27ijd31l8qgxkv5rv1z0c3hs7pn-python3.11-pytest-7.4.2/lib/python3.11/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
rpy2/tests/rinterface/test_bufferprotocol.py:4: in <module>
    rinterface.initr()
rpy2/rinterface.py:1082: in initr_checkenv
    _setrenvvars(_DEFAULT_ENVVAR_ACTION)
rpy2/rinterface.py:1156: in _setrenvvars
    for k, v in _getrenvvars():
rpy2/rinterface.py:1144: in _getrenvvars
    for k, v in reader:
E   ValueError: not enough values to unpack (expected 2, got 1)
```

**Comment by alejandrocorbellini-ac** [comment]
hi! we are also hitting this issue, doesn't this also break if the any of the environment variables in the system contains a comma?

**Comment by collares** [comment]
> hi! we are also hitting this issue, doesn't this also break if any of the environment variables in the system contains a comma?

I think R surrounds each value by quotes, and commas inside quotes should be fine. But I'm not really an R use, @lgautier certainly knows better.

**Comment by alejandrocorbellini-ac** [comment]
but at this point R is not involved, right? 
https://github.com/rpy2/rpy2/blob/f8770064227b2ea5695d8627a4d1476f6d39361a/rpy2/rinterface.py#L1143

in our case, we have an escaped JSON string in one of the environment variables in the system (totally unrelated to R env vars)

**Comment by collares** [comment]
You're right, commas in environment variables cause a "ValueError: too many values to unpack (expected 2)" error in this test.

**Comment by lgautier** [comment]
I see that I use R's `write.table()`, not `write.csv()`, and on the Python side read the output using the package `csv`. R's write.csv()` might escape sufficiently well the characters creating trouble.

**Comment by lgautier** [comment]
Can someone confirm that PR #1070 fixes the issue?

**Comment by lgautier** [comment]
Fixed with PR #1070 . The fix will be included in release 3.5.15.
